src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -92,6 +92,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Reference Include="System.Xml"/>
   </ItemGroup>
 
+  <!-- Set the default versions of the NETStandard.Library or Microsoft.NETCore.App packages to reference.
+       Choose the highest of the target framework version and the highest version of the package that the sdk knows about.
+       The implicit package references themselves are defined in Microsoft.NET.Sdk.props, so that they can be overridden by
+       including them in the project file. -->
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' And '$(NetStandardDefaultPackageVersion)' == ''">
+    <NetStandardDefaultPackageVersion>1.6</NetStandardDefaultPackageVersion>
+    <NetStandardDefaultPackageVersion Condition="'$(_TargetFrameworkVersionWithoutV)' > '$(NetStandardDefaultPackageVersion)' ">$(_TargetFrameworkVersionWithoutV)</NetStandardDefaultPackageVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(NetCoreAppDefaultPackageVersion)' == ''" >
+    <NetCoreAppDefaultPackageVersion>1.0.1</NetCoreAppDefaultPackageVersion>
+    <NetCoreAppDefaultPackageVersion Condition="'$(_TargetFrameworkVersionWithoutV)' > '$(NetCoreAppDefaultPackageVersion)' ">$(_TargetFrameworkVersionWithoutV)</NetCoreAppDefaultPackageVersion>
```

Comment:
Yes.  When you do a greater/less than comparison with strings, MSBuild tries to parse them as Versions and compare them as such.